boot/assets.go

```diff
@@ -354,6 +358,32 @@ func (o *TrustedAssetsInstallObserver) ChosenEncryptionKeys(key, saveKey keys.En
 	o.saveEncryptionKey = saveKey
 }
 
+func (o *TrustedAssetsInstallObserver) UpdateBootEntry() error {
+	if o.seedBootloader == nil {
+		return nil
+	}
+	efiBl, ok := o.seedBootloader.(bootloader.UefiBootloader)
+	if !ok {
+		return nil
+	}
+
+	var updatedAssets []string
+	for name := range o.trackedRecoveryAssets {
+		updatedAssets = append(updatedAssets, name)
+	}
+
+	description, assetPath, optionalData, err := efiBl.EfiLoadOptionParameters(updatedAssets)
+	if err != nil {
+		return fmt.Errorf("cannot get EFI load option parameter: %v", err)
+	}
+	if assetPath != "" {
+		if err := SetEfiBootVariables(description, assetPath, optionalData); err != nil {
```

Comment:
Potentially, should the check `if assetPath != ""` occur within `SetEfiBootVariables`, so that each caller doesn't need to make the check? I think `SetEfiBootVariables` currently assumes `assetPath` is non-nil, might be safer if it were the one to check.